compiler/runtime/OMRAOTRelocationRuntime.hpp

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_AOT_RELOCATION_RUNTIME_INCL
+#define OMR_AOT_RELOCATION_RUNTIME_INCL
+
+#ifndef OMR_AOT_RELOCATION_RUNTIME_CONNECTOR
+#define OMR_AOT_RELOCATION_RUNTIME_CONNECTOR
+namespace OMR { class OMRAOTRelocationRuntime; }
+namespace OMR { typedef OMR::OMRAOTRelocationRuntime OMRAOTRelocationRuntimeConnector; }
+#endif
+
+#include "runtime/TRRelocationRuntime.hpp"
+#include <map>
+
+namespace TR { class RelocationRuntime; }
+
+namespace OMR 
+{
+
+class OMR_EXTENSIBLE OMRAOTRelocationRuntime : public TR::RelocationRuntime 
```

Comment:
Maybe this has already been addressed in previous reviews, but if `OMRAOTRelocationRuntime` is marked extensible, why is it extending `TR::RelocationRuntime`? Should the `TR` namespace class be the most derived type?

Reply:
Hi, Thank you for your comment! This is a class inheriting from the extensible class, not most derived extensible class. We've figured during our previous discussion that mixing static and dynamic polymorphism can happen , and I think this is the use case. (Every AOT might need Relocation runtime and can consume it, but not every relocation runtime is AOT relocation runtime)

Reply:
I see, `OMRAOTRelocationRuntime` is designed to be it's own extensible class, which happens to inherit from another extensible class without actually extending it in the extensible framework.

Reply:
Right!

Reply:
I'm pretty sure it's resolved.